### PR TITLE
ResultProperties: Added keywords for CCSD quantities.

### PR DESCRIPTION
**Pull request by sjrl**
Added keywords for CCSD quantities. Is this all that needs to be done? Or does the qc_schema also need to be updated?

### CODE REVIEW

**Comment by dgasmith** [comment]
I think this should a PR to QCSchema first. A few of those quantities will need string explanations with them and there will be some feedback there I am guessing.

**Comment by sjrl** [comment]
I have made a PR to QCSchema for the CCSD quantities (https://github.com/MolSSI/QCSchema/pull/61).

**Comment by dgasmith** [comment]
This should be good to update.

**Comment by sjrl** [comment]
So I have added the CC keywords to QCElemental. Should I also go ahead with the renaming of `mp2_total_correlation_energy` to `mp2_correlation_energy`?

**Comment by loriab** [comment]
> Should I also go ahead with the renaming of mp2_total_correlation_energy to mp2_correlation_energy?

yes, please. any and all changes to make qcel agree with the qcschema PR.

**Comment by sjrl** [comment]
Okay I've updated the mp2 to match the QCSchema PR. Presumably this will cause some problems in other parts of the QCArchive.

File: qcelemental/models/results.py [hunk 1]
```diff
@@ -37,8 +37,36 @@ class ResultProperties(BaseModel):
...
+    # CCSD(T) keywords
+    ccsd_prt_pr_correlation_energy: float = None
+    ccsd_prt_pr_total_energy: float = None
+    ccsd_prt_pr_dipole_moment: List[float] = None
+
+    # CCSDT keywords
+    ccsdt_correlation_energy: float = None
+    ccsdt_total_energy: float = None
+    ccsdt_dipole_moment: List[float] = None
+    ccsdt_iterations: int = None
```
Comment (dgasmith) [inline qcelemental/models/results.py]: CCSDT(Q) last one (ugh).

Comment (sjrl) [reply qcelemental/models/results.py]: Is this for completions sake? Since it looks like there a number of other CC methods out there too.

Comment (dgasmith) [reply qcelemental/models/results.py]: Yea, honestly scratching my head on this one a bit after seeing it in practice. Anything beyond CCSD(T) might be better served in `qcvars`. @loriab?

Comment (loriab) [reply qcelemental/models/results.py]: just checked and CCSDT(Q) wasn't added to qcschema. I think it's ok to skip for now.

Comment (sjrl) [reply qcelemental/models/results.py]: Yeah it doesn't seem there are a lot of codes out there that support past CCSD(T). I guess if we are still on this topic I'd only advocate for the addition of CC-F12 methods.

Comment (loriab) [reply qcelemental/models/results.py]: in my experience, [these](https://github.com/qcdb/qcdb/blob/master/qcdb/psivarrosetta.py#L158-L183) are what you want to collect for f12 methods from molpro. not sure what balance of work-with-this-and-learn-from-practice or add-em-now to suggest.

Comment (loriab) [reply qcelemental/models/results.py]: oh, and one shouldn't actually be storing (T), (T*), (T**), as those can be built by equation

**Comment by sjrl** [comment]
I'm happy to leave as is and add in more in later rounds. So to clarify do I need to add a line for the current PR to https://github.com/MolSSI/QCElemental/blob/master/docs/source/changelog.rst ?

**Comment by dgasmith** [comment]
Lets leave it as is, but snip off everything after CCSD(T). The changeling line would be great!

**Comment by sjrl** [comment]
Okay so snip out stuff after CCSD(T) here in `results.py` but leave those quantities in QCSchema? And great I'll add a line.

**Pull request merged**